src/Elcodi/Component/Sitemap/Builder/SitemapBuilder.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of the Elcodi package.
+ *
+ * Copyright (c) 2014-2015 Elcodi.com
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Feel free to edit as you please, and have fun.
+ *
+ * @author Marc Morera <yuhu@mmoreram.com>
+ * @author Aldo Chiecchia <zimage@tiscali.it>
+ * @author Elcodi Team <tech@elcodi.com>
+ */
+
+namespace Elcodi\Component\Sitemap\Builder;
+
+use Elcodi\Component\Sitemap\Element\Interfaces\SitemapElementGeneratorInterface;
+use Elcodi\Component\Sitemap\Renderer\Interfaces\SitemapRendererInterface;
+
+/**
+ * Class SitemapBuilder
+ */
+class SitemapBuilder
+{
+    /**
+     * @var SitemapElementGeneratorInterface[]
+     *
+     * sitemapElementGenerators
+     */
+    protected $sitemapElementGenerators;
+
+    /**
+     * @var SitemapRendererInterface
+     *
+     * sitemapRenderer
+     */
+    protected $sitemapRenderer;
+
+    /**
+     * @var string
+     *
+     * path
+     */
+    protected $path;
+
+    /**
+     * Construct
+     *
+     * @param SitemapRendererInterface $sitemapRenderer sitemapRenderer
+     * @param string                   $path            Path
+     */
+    public function __construct(
+        SitemapRendererInterface $sitemapRenderer,
+        $path
+    ) {
+        $this->sitemapRenderer = $sitemapRenderer;
+        $this->path = $path;
+    }
+
+    /**
+     * Add a new SitemapElement Generator
+     *
+     * @param SitemapElementGeneratorInterface $sitemapElementGenerator sitemapRenderer
+     *
+     * @return $this
+     */
+    public function addSitemapElementGenerator(
+        SitemapElementGeneratorInterface $sitemapElementGenerator
+    ) {
+        $this->sitemapElementGenerators[] = $sitemapElementGenerator;
+
+        return $this;
+    }
+
+    /**
+     * Build sitemap builder
+     *
+     * @param string|null $language Language
+     *
+     * @return string Generated data
+     */
+    public function build($language = null)
+    {
+        $sitemapElements = [];
+
+        /**
+         * @var SitemapElementGeneratorInterface $sitemapElementGenerator
```

Comment:
With [this typehint](https://github.com/elcodi/elcodi/pull/712/files#diff-ad3ac0922fdfce608f813798d73fc47cR29) this one should not be needed.